### PR TITLE
feat: handle pool updates

### PR DESCRIPTION
**Pull request by offerijns**

### CODE REVIEW

**Comment by offerijns** [comment]
Deployed to https://thegraph.com/explorer/subgraph/offerijns/tinlake-kovan-staging?version=pending

File: src/domain/Pool.ts [hunk 1]
```diff
@@ -69,3 +69,55 @@ export function createPoolHandlers(addresses: PoolAddresses): void {
...
+  context.setString('id', newAddresses.id)
+
+  log.debug('createUpdatedPoolHandlers: {} => {}', [prevAddresses.id, newAddresses.id])
+
+  if (prevAddresses.coordinator != newAddresses.coordinator) {
+    log.debug('createUpdatedPoolHandlers: swapping coordinator {} => {}', [
+      prevAddresses.coordinator,
+      newAddresses.coordinator,
+    ])
+    CoordinatorTemplate.createWithContext(Address.fromString(newAddresses.coordinator), context)
```
Comment (philipstanislaus) [inline src/domain/Pool.ts:84]: This does not swap them, it does only create the new handlers, right? Can we rename this log line and the ones below to: 

```suggestion
    log.debug('createUpdatedPoolHandlers: creating handler for changed coordinator {} => {}', [
      prevAddresses.coordinator,
      newAddresses.coordinator,
    ])
    CoordinatorTemplate.createWithContext(Address.fromString(newAddresses.coordinator), context)
```

File: src/mappings/PoolRegistry.ts [hunk 2]
```diff
@@ -36,6 +36,28 @@ export function handlePoolUpdated(call: PoolUpdated): void {
     call.params.name,
     call.params.data,
```
Comment (philipstanislaus) [inline src/mappings/PoolRegistry.ts:37]: Please update the comment above

File: src/mappings/PoolRegistry.ts [hunk 3]
```diff
@@ -36,6 +36,28 @@ export function handlePoolUpdated(call: PoolUpdated): void {
...
   ])
+  let poolId = call.params.pool.toHexString()
+  let hash = call.params.data
+  let oldPoolAddresses = PoolAddresses.load(poolId)
+
+  if (oldPoolAddresses == null) {
+    log.error('handlePoolUpdated: could not load old pool addresses', [])
+    return
```
Comment (philipstanislaus) [inline src/mappings/PoolRegistry.ts:45]: Should this not created the handlers anyways? There is no cost to us to create them, right?

File: src/domain/PoolAddresses.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { TypedMap, JSONValue } from '@graphprotocol/graph-ts'
 import { PoolAddresses } from '../../generated/schema'
 
-export function createPoolAddresses(poolId: string, addresses: TypedMap<string, JSONValue>): PoolAddresses {
+export function updatePoolAddresses(poolId: string, addresses: TypedMap<string, JSONValue>): PoolAddresses {
   let poolAddresses = new PoolAddresses(poolId)
```
Comment (philipstanislaus) [inline src/domain/PoolAddresses.ts:5]: Should this not load the addresses first and only create the pool if addresses are not found?

**Review by philipstanislaus — COMMENTED** [review]
Clean and nice. I feel that we should definitely test this, not 100% sure though how. 

Ideally tinlake-in-a-box with a local graph node in a docker container, and then use a dapp tools to send txs and query the graph? Sounds like this will take a few days to setup, but it would be quite some risk if the swapping of pools goes wrong. Simply redeploying would not help much either, since the ingestion will take forever.

What do you think?

**Pull request merged**